### PR TITLE
add UT for pkg/apis/autoscaling/v2alpha1/defaults.go

### PR DESCRIPTION
**Pull request by fisherxu**
**What this PR does / why we need it**:
add UT for pkg/apis/autoscaling/v2alpha1/defaults.go

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by dims** [comment]
@fisherxu i'd recommend squashing the 2 commits into one

**Comment by fisherxu** [comment]
Thanks :) @dims  have done!

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		},
+	}
+
+	for i, test := range tests {
+		original := test.original
+		expected := test.expected
+		obj2 := roundTrip(t, runtime.Object(original))
+		got, ok := obj2.(*autoscalingv2alpha1.HorizontalPodAutoscaler)
+		if !ok {
+			t.Fatalf("(%d) unexpected object: %v", i, got)
```
Comment (dims) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: Should we print `obj2` instead of `got`?

Comment (dims) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: Also is `Errorf` more appropriate than `Fatalf`?

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+func roundTrip(t *testing.T, obj runtime.Object) runtime.Object {
+	data, err := runtime.Encode(api.Codecs.LegacyCodec(SchemeGroupVersion), obj)
+	if err != nil {
+		t.Errorf("%v\n %#v", err, obj)
+		return nil
+	}
+	obj2, err := runtime.Decode(api.Codecs.UniversalDecoder(), data)
+	if err != nil {
+		t.Errorf("%v\nData: %s\nSource: %#v", err, string(data), obj)
```
Comment (dims) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: In line 110 and 121 we print the actual object, why are we doing a `string(data)` here?

**Comment by dims** [comment]
@fisherxu this is new area for me to look at, so just added some questions to figure it out

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go (hunk 1 above)
Comment (fisherxu) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: Thanks for review, **1)** it should print obj2 , will fix it. **2)**Fatalf has same effect as Errorf and FailNow, only Errorf will continue execution.

Comment (fisherxu) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: In pkg/apis/autoscaling/v1/defaults_test.go, it also use the Fatalf  :)

Comment (dims) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: Great! thanks

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go (hunk 2 above)
Comment (fisherxu) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: data is []byte type here, here just print the data in string will more user-friendly, and the roundTrip function in every pkg/apis/GROUPS is same.

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go (hunk 1 above)
Comment (fisherxu) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: @dims  have addressed it, print obj2 now!

File: pkg/apis/autoscaling/v2alpha1/defaults_test.go (hunk 2 above)
Comment (dims) [inline pkg/apis/autoscaling/v2alpha1/defaults_test.go]: Ack thanks!

**Comment by fisherxu** [comment]
@deads2k  Can you help review this pr? :)

**Comment by fisherxu** [comment]
@sttts  Because just add UT, I don't create new issue. Can you help add **approve no-issue** flag? Thanks

**Comment by fisherxu** [comment]
@dims  Please help add **approve no-issue** flag, thanks~ :)

**Comment by dims** [comment]
@fisherxu unfortunately i don't have authority to do that. you still need someone from https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/OWNERS to approve before someone can apply `lgtm`

**Pull request merged**